ansible/roles/netbox/tasks/main.yml

```diff
@@ -220,6 +220,24 @@
   register: netbox__register_migration
   changed_when: netbox__register_migration.changed | bool
 
+- name: Check if Django superuser account exists
+  # In case of an upgrade (with DB migrations), the superuser will
+  # be existent but the createsuperuser command will run anyway (and fails).
+  # So we check if it exist beforehand and only run the create command if needed.
+  environment:
+    DJANGO_SUPERUSER_PASSWORD: '{{ netbox__superuser_password }}'
+  django_manage:
+    command: shell -c 'import sys; from django.contrib.auth.models import User; print(User.objects.filter(username="{{ netbox__superuser_name }}").count())'
+    app_path: '{{ netbox__git_checkout + "/netbox" }}'
+    virtualenv: '{{ netbox__virtualenv }}'
+  become: True
+  become_user: '{{ netbox__user }}'
+  when: netbox__primary|bool
+  no_log: '{{ debops__no_log | d(True) }}'
+  register: netbox__check_superuser
+  ignore_errors: true
```

Comment:
I try not to add `ignore_errors: True` in DebOps code. Any error should be accounted for and handled appropriately.

Did you encounter any errors while testing this task? Does it work when the superuser account doesn't exist yet?

Reply:
I actually don't recall the reason on why this is required. But I do recall having some trouble with the jango part, as it was not idempotent at all.
Do you want me to check this again and verify if it is required or not?

Reply:
Yes please, if you can. It could be an issue in your environment, but then if we can replicate it, we could try and find the cause.

Reply:
A few changes were merged recently in the `netbox` role, you might want to check them out.

Reply:
@drybjed The following commit (https://github.com/debops/debops/commit/a6896f579f83cf2d14cc1d12a1600f41e57fedfc) is actually doing the same but different.  From looking at the code only, I'd say this should resolve the initial issue I tried fixing in this PR.

We can close this one.